### ISSUE TITLE
Android flickering when passing in value prop

### ISSUE THREAD

**Issue by Jpoliachik** [react-native-community/react-native-text-input-mask#11]
Passing in a value on Android causes a noticeable flicker. 
Reproduce by using state to update value onChange and pass into TextInputMask on each render. 

Video of the issue: https://youtu.be/b0aTUZtd-t4

**Comment by Jpoliachik** [comment]
For additional clarity - it's easier to reproduce when the TextInputMask is on a complex view. Try adding 10 TextInputMask in one view, and it'll flicker heavily.

**Comment by ivanzotov** [comment]
@Jpoliachik thanks for the issue, will try to reproduce and fix

**Comment by Jpoliachik** [comment]
@ivanzotov thanks for the response. I've been experimenting more, and I've noticed a possible related issue when passing in unformatted `value` prop: 

```
<TextInputMask
          onChangeText={(formatted, extracted) => {
            this.setState({ text: extracted });
          }}
          value={this.state.text}
          mask={"+1 ([000]) [000] [00] [00]"}
        />
```
It flickers as formatted for a frame, then reverts to the unformatted numbers. Take a look.

**Comment by ivanzotov** [comment]
Fixed in 0.5.2 by @BasitAli (#18)

**Comment by andrewcroce** [comment]
Something similar seems to be occurring, even with only setting a `defaultValue` and leaving the value uncontrolled.

Backspacing behavior jumps back several characters at a time.

Using:
react 16.0.0
react-native 0.52.0
react-native-text-input-mask 0.5.2


```
<TextInputMask
  defaultValue={ this.props.value }
  onChangeText={ this.handleChange }
  mask: '[0000] - [0000] - [0000] - [0000]' />

...

handleChange = (formattedValue, extractedValue) => {
  // Im not actually modifying the field's state here
}
```
![android-mask](https://user-images.githubusercontent.com/474639/36174161-65c2cf48-10d9-11e8-99ae-9c2f13224194.gif)
